policy/aggregation_id_compress.go

```diff
@@ -35,7 +35,7 @@ type AggregationIDCompressor interface {
 type AggregationIDDecompressor interface {
 	// Decompress decompresses aggregation types,
 	// returns error if any invalid aggregation type is encountered.
-	Decompress(compressed AggregationID) (AggregationTypes, error)
+	Decompress(pool AggregationTypesPool, compressed AggregationID) (AggregationTypes, error)
```

Comment:
Shouldn't the pool be in the implementation instead? i.e., in the ctor:
`NewAggregationIDDecompressor(pool AggregationTypePool)`.

Then you still have `Decompress(compressed AggregationID)`.

And when you are done with it, return it to the pool?